### PR TITLE
[WIP] New component designs and styleguide: https://sketch.cloud/s/3nq5w

### PR DESCRIPTION
**Pull request by sqevani**

### CODE REVIEW

File: build/webpack.conf.js [hunk 1]
```diff
@@ -1,65 +1,176 @@
...
+        return module.context && module.context.includes('node_modules')
+      }
+    })
   ],
-};
+  resolve: {
+    alias: {
+      // Global paths
+      '@source': path.resolve(__dirname, '../src'),
+      '@public': path.resolve(__dirname, '../public'),
```
Comment (davidyuk) [inline build/webpack.conf.js]: Why don't use absolute paths (`/src/...` and `/public`) instead of this aliases?

File: package.json [hunk 2]
```diff
@@ -23,9 +23,11 @@
   "dependencies": {
     "ethereum-blockies-png": "git+https://github.com/acdbaykal/ethereum-blockies-png.git",
     "ethjs-unit": "^0.1.6",
+    "normalize-scss": "^7.0.1",
```
Comment (davidyuk) [inline package.json]: Why should we replace `normalize.css` with this package?

File: styleguide.config.js [hunk 3]
```diff
@@ -37,8 +37,8 @@ module.exports = {
...
     }
   },
-  require: [
-    path.join(__dirname, 'src/components/common.scss')
-  ],
-  assetsDir: 'static'
+  //require: [
+  //  path.join(__dirname, 'src/components/common.scss')
+  //],
+  assetsDir: 'public'
```
Comment (davidyuk) [inline styleguide.config.js]: What is the reason for renaming `static` folder?

File: src/components/aeInput/index.js [hunk 4]
```diff
@@ -1,5 +1,8 @@
-import aeInput from './aeInput.vue'
-
-export default function install (Vue) {
-  Vue.component('ae-input', aeInput)
+/**
+ * aeInput instance
+ * @param {Vue} Vue
+ * @param {{}} options
+ */
+export const install = function (Vue, options) {
```
Comment (davidyuk) [inline src/components/aeInput/index.js]: `options` is the unused argument, why you have added it?

**Comment by sqevani** [comment]
Explanation to what's happening with the big bundle:

- Refactored Webpack and Stylguidist

Webpack now will chunk several things, reduced code generation, minification and other related stuff.

Currentl webpack spits:

1. aepp.components.js, aepp.components.css
- these two are the entire library of components available, and if someone wants to use them simply, imports them and data. Everything is there.

2. aepp.base.css 
- This one is simply the root level CSS that is needed overall for each component to work separately

3. aepp.base.js is just a random file, since which needs to be removed everytime there is a build. (will work on this later.

4. FUTURE: There will be a .js output for each component, and making use of aepp.base.css each component can work independently.

Changed the directories names:

- dist/ -> build/
- build/ -> config/
- created docs/ folder for documentation

Renamed several npm run-scripts names to follow a specific pattern

- `verb`
- `noun` (optional)
- `noun:verb`

```
    "unit": "jest --config config/jest.config.js",
    "unit:watch": "jest --watch",
    "lint": "eslint --ext .js,.vue src",
    "test": "npm run unit && npm run lint",
    "build": "node config/build.js",
    "watch": "node config/watch.js",
    "docs": "vue-styleguidist server --verbose --config config/styleguidist.config.js",
    "docs:build": "vue-styleguidist build --verbose --config config/styleguidist.config.js",
    "prepublish": "npm test",
    "prepare": "npm run build"
```

Also, when it comes to Environmental variables, there's one place to define those. its `.env` and `dotenv` plugin takes care of all the rest.

----

Oh also, size went from: 400KB -> 200KB unminified, minified as for minified it is (drum roll):

screen shot 2018-07-12 at 18 30 36

File: .eslintrc.js [hunk 5]
```diff
@@ -18,11 +18,9 @@ module.exports = {
...
     // don't require .vue extension when importing
-    'import/extensions': ['error', 'always', {
-      'js': 'never',
-      'vue': 'never'
-    }],
+    'import/extensions': false,
     // allow debugger during development
-    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0
+    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
+    "indent": "off"
```
Comment (davidyuk) [inline .eslintrc.js]: This rule should be enabled.

File: .eslintrc.js [hunk 6]
```diff
@@ -18,11 +18,9 @@ module.exports = {
   // add your custom rules here
   'rules': {
     // don't require .vue extension when importing
-    'import/extensions': ['error', 'always', {
-      'js': 'never',
-      'vue': 'never'
-    }],
+    'import/extensions': false,
```
Comment (davidyuk) [inline .eslintrc.js]: What is the reason for this change?

File: .githooks/pre-commit/lint [hunk 7]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+npm run lint
```
Comment (davidyuk) [inline .githooks/pre-commit/lint]: I'm against stuff like this: https://github.com/aeternity/aepp-identity/pull/82

Comment (sadiqevani) [reply .githooks/pre-commit/lint]: I would like to catch most of the issues before committing and/or pushing.

File: .eslintrc.js (hunk 6 above)
Comment (sadiqevani) [reply .eslintrc.js]: its like that for now, my IDE had some issues as it didnt recognize the exceptions

File: .githooks/pre-commit/lint (hunk 7 above)
Comment (davidyuk) [reply .githooks/pre-commit/lint]: So, I'm repeating some points from https://github.com/aeternity/aepp-identity/pull/82:
- it is not working properly when developer trying to commit part of the changes of the working directory
- It slows down git performance
- these checks are already performs in CI

File: src/components/ae-toolbar/ae-toolbar.html [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<div class="ae-toolbar" :class="{ [mode]: Boolean(mode), [align]: Boolean(align), minimal, disabled }">
```
Comment (davidyuk) [inline src/components/ae-toolbar/ae-toolbar.html]: `mode` and `align` won't be casted to Boolean automatically?

File: src/components/ae-toolbar/ae-toolbar.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+     * Select
+     */
+    mode: {
+      type: String,
+      validator: function (value) {
+        return [
+          'primary',
+          'secondary'
+        ].indexOf(value) !== -1
+      }
```
Comment (davidyuk) [inline src/components/ae-toolbar/ae-toolbar.js]:
```
validator: value => ['primary', 'secondary'].includes(value)
```
?
